### DISCUSSION TITLE
Arbitrary variants adding unwanted prefix

### DISCUSSION THREAD

**Discussion by michael-tugby** [tailwindlabs/tailwindcss#8932] [Help]
Hi,

Attempting to upgrade to the latest version of Tailwind, but I'm hitting a little issue with the new arbitrary variants adding prefixes to classes.

For example, if I do `[&>.child]`, if I only use this variant once, it's fine. But, every other time I use that variant in the same class list, it will add an unwanted prefix to the start.

Example[ here.](https://play.tailwindcss.com/PCNcsYeV8M)

I'm also aware of a recent bugfix that was done to this [here](https://github.com/tailwindlabs/tailwindcss/issues/8626), which is probably related?

**Comment by michael-tugby** [comment]
Just had a look at that PR, and I think what may be happening is that after the first pass, the custom variant gets added to a map. So, on the second pass, it finds the variant in the map, hence it skips the logic to toggle `isArbitraryVariant` to `true`.

Saying that though, I don't have a good understanding of the code behind tailwind, so my assumption here may be off.

**Reply by wongjn** [reply]
Your analysis looks right, I'm working on a PR based on your findings.

Edit: PR is #8992
